src/main/java/com/dmz/airdnd/accommodation/service/AddressService.java

```diff
@@ -0,0 +1,47 @@
+package com.dmz.airdnd.accommodation.service;
+
+import java.util.Optional;
+
+import org.locationtech.jts.geom.Point;
+import org.springframework.stereotype.Service;
+
+import com.dmz.airdnd.accommodation.adapter.GeocodingClient;
+import com.dmz.airdnd.accommodation.domain.Address;
+import com.dmz.airdnd.accommodation.dto.response.CoordinatesDto;
+import com.dmz.airdnd.accommodation.repository.AddressRepository;
+import com.dmz.airdnd.accommodation.util.GeoPointFactory;
+
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@Service
+public class AddressService {
+
+	private final AddressRepository addressRepository;
+	private final GeocodingClient geocodingClient;
+
+	public Address getOrCreateByFullAddress(String country, String baseAddress, String detailedAddress) {
+		String detail = Optional.ofNullable(detailedAddress)
+			.map(String::trim)
+			.orElse("");
```

Comment:
Optional은 값이 있을 수도 있고 없을 수도 있다를 표현하기 위한 컨테이너인데, 지금 로직으로는 String Null 체크에 쓰이고 있어요.
보통은 Optional한 값이 올수 있다는것을 의미하도록 메서드 리턴값에만 쓰는 게 좋습니다. 특히 String은 원래 nullable한 객체이므로 굳이 Optional로 감쌀 필요는 없습니다. null 체크가 목적이면 if로 체크하거나 

```suggestion
		String detail = StringUtils.trimToEmpty(detailedAddress);
```
요런 string util을 사용해도 좋습니다.

Reply:
찾아보니 trimToEmpty()는 입력이 null이면 빈 문자열을 반환하고, null이 아니면 앞뒤 공백을 제거한 값을 반환하네요.
좋은 방법인 것 같습니다. 감사합니다👍